sql-plugin/src/main/scala/org/apache/spark/sql/rapids/collectionOperations.scala

```diff
@@ -103,21 +104,55 @@ case class GpuElementAt(left: Expression, right: Expression)
     lhs.dataType match {
       case _: ArrayType => {
         if (rhs.isValid) {
-          val ordinalValue = rhs.getValue.asInstanceOf[Int]
-          if (ordinalValue > 0) {
-            // SQL 1-based index
-            lhs.getBase.extractListElement(ordinalValue - 1)
-          } else if (ordinalValue == 0) {
+          if (rhs.getValue.asInstanceOf[Int] == 0) {
             throw new ArrayIndexOutOfBoundsException("SQL array indices start at 1")
-          } else {
-            lhs.getBase.extractListElement(ordinalValue)
+          } else  {
+            // Positive or negative index
+            val ordinalValue = rhs.getValue.asInstanceOf[Int]
+            withResource(lhs.getBase.countElements) { numElementsCV =>
+              withResource(numElementsCV.min) { minScalar =>
+                val minNumElements = minScalar.getInt
+                // index out of bound
+                // Note: when the column is containing all null arrays, CPU will not throw, so make
+                // GPU to behave the same.
+                if (minNumElements < math.abs(ordinalValue) &&
+                  lhs.getBase.getNullCount != lhs.getBase.getRowCount &&
+                  failOnError) {
```

Comment:
nit: The below can avoid the evaluation of the follwoing two expressions if `failOnError ` is false.
```Scala
                if (failOnError &&
                  minNumElements < math.abs(ordinalValue) &&
                  lhs.getBase.getNullCount != lhs.getBase.getRowCount) {
```